numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -3037,6 +3037,13 @@ array_arange(PyObject *NPY_UNUSED(ignored),
         }
     }
 
+    if (typecode != NULL && (typecode->kind == 'S' || typecode->kind == 'U')) {
```

Comment:
This looks okay to me, but I'm not sure what the recommended way for test for string/bytes dtypes is - @seberg or @mattip is this how you'd do it?

Reply:
I would use `PyDataType_ISSTRING`.  But, I think it may also be better to move it into `PyArray_ArangeObj`.  There may even be a good place so that `np.arange("a", "b")` gives the same error.

Reply:
I moved the check to `PyArray_ArangeObj`, and it supports `np.arange("a", "b")` now too. I added a test for that as well.

Reply:
Should this added code be removed now that the check lives elsewhere?